common/src/main/java/com/wynntils/utils/render/MapRenderer.java

```diff
@@ -146,6 +150,265 @@ public static void renderCursor(
         }
     }
 
+    public static void renderLootrunLine(
+            LootrunInstance lootrun,
+            float lootrunWidth,
+            PoseStack poseStack,
+            float centerX,
+            float centerZ,
+            float mapTextureX,
+            float mapTextureZ,
+            float currentZoom,
+            int color) {
+        if (lootrun.simplifiedPath().size() < 3) return;
+
+        BufferBuilder bufferBuilder = Tesselator.getInstance().getBuilder();
+        bufferBuilder.begin(VertexFormat.Mode.TRIANGLES, DefaultVertexFormat.POSITION_COLOR);
+        RenderSystem.setShader(GameRenderer::getPositionColorShader);
+        RenderSystem.setShaderColor(1.0f, 1.0f, 1.0f, 1.0f);
+        RenderSystem.disableCull();
+
+        List<Vector3f> points = new ArrayList<>();
+
+        List<Vector3f> middlePoints = new ArrayList<>();
+
+        for (Vector3f point : lootrun.simplifiedPath()) {
+            points.add(new Vector3f(
+                    getRenderX((int) point.x(), mapTextureX, centerX, currentZoom),
+                    getRenderZ((int) point.z(), mapTextureZ, centerZ, currentZoom),
+                    0));
```

Comment:
Shouldn't it be x, 0, z? (Again, does this strictly need to be a Vector3f if we have a filler element?)

Reply:
I am more worried about this allocation since this is render time, than the lootrun compiling one.

Reply:
I agree that this is doing scary much during rendering. All this processing, can't it be done outside rendering? Ideally, we should know exactly what we need to render when it is time to render, and just do that as fast as possible with as few allocations as possible.

Reply:
as long as it still need screenspace transformation it will need to be in render thread, an object pool maybe?

Reply:
> I agree that this is doing scary much during rendering. All this processing, can't it be done outside rendering? Ideally, we should know exactly what we need to render when it is time to render, and just do that as fast as possible with as few allocations as possible.

Perhaps I could profile this. I know it looks like a lot of code, but it is actually rather fast compared to what legacy does for example... Adding more vertexes (or rather not filtering them) is a LOT more slower than this whole process. I will come back here soon with some results.

Reply:
Legacy didn't optimise at all and just always rendered the whole path, so that's like a worst case or hard limit comparison

Reply:
Yes, I know. But having actual numbers will either make me and magicus relax, or we can find the problem. Win-Win situation.

Reply:
True - it wasn't my intention to argue against profiling (rather the opposite), just to point out that Legacy isn't a good comparison

Reply:
Okay. Performance numbers. 97% of the time we spend is actual rendering, around 3% is the math. So I think this concludes the need for optimization.

Reply:
@Kepler-17c I agree that legacy is not a good comparison. But our bar for Artemis is higher than replicating legacy.

@kristofbolyai Thanks for the measuring. I'm not sure I understand what fix you are pointing to, though...

Reply:
@magicus https://github.com/Wynntils/Artemis/pull/1688#discussion_r1257100736